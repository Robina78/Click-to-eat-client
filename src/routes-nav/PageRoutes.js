import React from 'react';
import { Routes, Route, Navigate } from "react-router-dom";
import LoginForm from '../auth/LoginForm';
import SignupForm from '../auth/SignupForm';
import Homepage from '../homepage/Homepage';
import ProfileForm from '../profile/ProfileForm';
import Restaurants from '../Restaurant/Restaurants';
import PrivateRoute from './PrivateRoute';
import PostRestaurant from '../PostedRestaurants/PostRestaurant';
import RestaurantList from '../PostedRestaurants/RestaurantList';

export default function PageRoutes({ login, signup}) {
    return (
        <div className="pageRoutes">
            <Routes>

                <Route exact path="/" element={<Homepage />} />
                <Route exact path="/login" element={<LoginForm login={login}/>} />
                <Route exact path="/signup" element={<SignupForm  signup={signup}/>} />

                <Route exact 
                    path="/profile" 
                    element={
                        <PrivateRoute>
                            <ProfileForm />
                        </PrivateRoute>
                    }
                />

                <Route exact 
                    path="/restaurants" 
                    element={
                        <PrivateRoute>
                            <Restaurants />
                        </PrivateRoute>
                    }
                />
                <Route exact 
                    path="/post" 
                    element={
                        <PrivateRoute>
                            <PostRestaurant />
                        </PrivateRoute>
                    }
                />
                <Route exact 
                    path="/list" 
                    element={
                        <PrivateRoute>
                            <RestaurantList />
                        </PrivateRoute>
                    }
                />

                <Route element={<Navigate to="/" />}/>                
            </Routes>            
        </div>
    )
}

