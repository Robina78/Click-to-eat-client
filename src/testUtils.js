import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
    username: "testuser",
    first_name: "testfirst",
    last_name: "testlast",
    email: "test@gmail.com",
    phone: "233466",
    profile_photo:  null,
};

const UserProvider =
 ({ children, currentUser = demoUser}) => (
     <UserContext.Provider value={{ currentUser}}>
         {children}
     </UserContext.Provider>
 );

 export { UserProvider};