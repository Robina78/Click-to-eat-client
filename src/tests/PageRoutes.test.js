import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";
import PageRoutes from "../routes-nav/PageRoutes";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <PageRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <PageRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
