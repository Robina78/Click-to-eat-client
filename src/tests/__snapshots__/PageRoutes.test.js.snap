// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pageRoutes"
  >
    <div
      class="Homepage container-fluid"
    >
      <div
        class="wrapper"
      >
        <h1>
          Hi testuser 
        </h1>
        <h3>
          <span />
          <span
            class="ityped-cursor"
          >
            |
          </span>
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
